### DISCUSSION TITLE
pet exec always uses /bin/sh?

### DISCUSSION THREAD

**Discussion by marcopaganini** [knqyf263/pet#313] [General]
Hello,

Looking at the code in:

https://github.com/knqyf263/pet/blob/b6ecc509a60e979652948b7ffaa6bd5e13c3029f/cmd/util_unix.go#L19

It appears that `pet exec` always uses `/bin/sh -c` to execute commands. In many linux systems, `/bin/sh`  is symlinked to `/bin/dash` which is the basic bourne shell. Since I use bash (and most don't use dash, I suppose), does it make sense to execute commands using `/bin/sh`? Many of my one-liners use bash constructs, so this makes pet unusable for those.

Any specific reasons why pet uses `/bin/sh` here?

**Comment by RamiAwar** [comment]
I feel like I've seen this same issue before somewhere but can't find it.

I think we can definitely extract this into global config. Some people might want different shells, makes sense!

**Comment by marcopaganini** [comment]
Thanks! One workaround for people using Debian based distributions is to
uninstall dash (`sudo apt remove --purge dash`). This makes bash the main
shell and things work again.

**Comment by RamiAwar** [comment]
Indeed this is configurable with the 'cmd' global config parameter! @marcopaganini 

Cmd = ["/bin/dash", "-c"]

or similar.
